wp-content/themes/pub/wporg-learn-2024/inc/block-config.php

```diff
@@ -557,3 +559,78 @@ function inject_other_filters( $key ) {
 		printf( '<input type="hidden" name="s" value="%s" />', esc_attr( $wp_query->query['s'] ) );
 	}
 }
+
+/**
+ * Filter the block data for the student course list.
+ *
+ * @param array $parsed_block The parsed block data.
+ * @return array The updated block data.
+ */
+function filter_student_course_list( $parsed_block ) {
+	if (
+		'core/query' !== $parsed_block['blockName'] ||
+		'course' !== ( $parsed_block['attrs']['query']['postType'] ?? '' ) ||
+		! isset( $parsed_block['attrs']['queryId'] ) ||
+		! is_page( 'my-courses' )
+	) {
+		return $parsed_block;
+	}
+
+	if ( ! array_key_exists( 'exclude', $parsed_block['attrs']['query'] ) || ! is_array( $parsed_block['attrs']['query']['exclude'] ) ) {
+		$parsed_block['attrs']['query']['exclude'] = array();
+	}
+
+	$exclude_ids = get_course_ids_to_be_excluded( $parsed_block['attrs']['queryId'] );
```

Comment:
Thinking about this further, since `active` and `completed` were working before, do we need to run this exclusion logic for them? It seems quite expensive as there are several queries and merges/diffs involved.

Reply:
Just tried this and it works for me locally. We could then simplify `get_course_ids_to_be_excluded` to only handle the `all` case.

```suggestion
        $filter_param_key = get_student_course_filter_query_var_name();
	$selected_option  = isset( $_GET[ $filter_param_key ] ) ? sanitize_text_field( wp_unslash( $_GET[ $filter_param_key ] ) ) : '';

	if ( ! empty( $selected_option ) && ( 'active' === $selected_option || 'completed' === $selected_option ) ) {
		return $parsed_block;
	}
	
	$exclude_ids = get_course_ids_to_be_excluded( $parsed_block['attrs']['queryId'] );
```

Reply:
Yeah, you are right. Although 'active' and 'completed' were working before, I am writing them here because I'd like to figure out their order first, so I haven't dealt with them. I also thought to first see if this method is a good approach before making further modifications.

7fb7b62
In the commit, I'm not just targeting 'all', but all statuses other than 'completed' and 'active'. This corresponds to the logic [here](https://github.com/Automattic/sensei/blob/trunk/includes/blocks/course-list/class-sensei-course-list-student-course-filter.php#L95), which means if it's not 'active' or 'completed', it shows all enrolled courses.

I'm not sure if the comment added is simple and clear enough, though..